halo2_frontend/src/dev.rs

```diff
@@ -1360,28 +1361,54 @@ mod tests {
                 let q = meta.complex_selector();
                 let table = meta.instance_column();
                 let advice_table = meta.advice_column();
+                let s_ltable = meta.complex_selector();
 
                 meta.annotate_lookup_any_column(table, || "Inst-Table");
                 meta.enable_equality(table);
                 meta.annotate_lookup_any_column(advice_table, || "Adv-Table");
                 meta.enable_equality(advice_table);
 
                 meta.lookup_any("lookup", |cells| {
+                    //
+                    // NOTE: When using `lookup_any` in circuit, the extra `Fixed` column or
+                    //      `Selector` is required to be enabled, in order to indicate the lookup table values.
+                    //
+                    //  The following diagram shows the example circuit.
+                    //  (This is NOT a real `FaultyCircuit` instantiation we have in this test. Just borrow of its config.)
```

Comment:
I'm a bit confused by this statement.  Could you clarify this?

Also I'm a bit confused about this test :sweat_smile:, it assigns two regions, and the table is assigned twice (once per region).  I think it would be more clear with 3 regions:
- 1 region to assign the table once
- 1 region to enable `q` and assign the inputs `a`: "good"
- 1 region to enable `q` without assignment of `a`: "faulty"


Anyway, I guess you updated this test so that the the new failure check doesn't trigger, right?

Reply:
Thanks for letting me know! @ed255 

> I'm a bit confused by this statement. Could you clarify this?

Yes, I see that the new test and example circuit explanation confuses people. 😅 
I will improve the statement.


> Anyway, I guess you updated this test so that the the new failure check doesn't trigger, right? 

Yes, I updated the test so that it does lead to wrong assignment error, not the `lookup_any` check error.